yarn-project/aztec-node/src/aztec-node/aztec-node.ts

```diff
@@ -94,6 +94,20 @@ export interface AztecNode {
    */
   getDataTreePath(leafIndex: bigint): Promise<SiblingPath<typeof PRIVATE_DATA_TREE_HEIGHT>>;
 
+  /**
+   * Find the index of the relevant l1 to l2 message.
+   * @param leafValue - The value to search for.
+   * @returns The index of the given leaf in the l1 to l2 message tree or undefined if not found.
+   */
+  findL1ToL2MessageIndex(leafValue: Buffer): Promise<bigint | undefined>;
```

Comment:
There could be a case that findL1ToL2MessageIndex could be combined with getL1ToL2Message?